src/persist/src/location.rs

```diff
@@ -373,6 +412,35 @@ impl From<(&str, &str)> for LockInfo {
     }
 }
 
+/// An abstraction over read-write access to a `bytes key`->`bytes value` store.
+///
+/// TODO: Rename this to Blob when we delete Blob.
+#[async_trait]
+pub trait BlobMulti: std::fmt::Debug {
+    /// Returns a reference to the value corresponding to the key.
+    async fn get(&self, deadline: Instant, key: &str) -> Result<Option<Vec<u8>>, LocationError>;
+
+    /// List all of the keys in the map.
+    async fn list_keys(&self, deadline: Instant) -> Result<Vec<String>, LocationError>;
+
+    /// Inserts a key-value pair into the map.
+    ///
+    /// When atomicity is required, writes must be atomic and either succeed or
+    /// leave the previous value intact.
+    async fn set(
```

Comment:
nit: should this be a `&mut self`? and same for `delete` below? Those two functions took mutable references in `Blob` before so not sure if this change is intentional or not

Reply:
that's intentional. see the comment thread where I made the same change to Consensus for why